models/klaviyo.yml

```diff
@@ -0,0 +1,276 @@
+      - name: numeric_value
+        description: Numeric value associated with the event (ie the dollars associated with a purchase).
+      - name: touch_type
+        description: >
+          Type of touch/message that the event itself is already attributed to in Klaviyo. Either 'flow', 'campaign', or null.
+          Note that the package will refer to campaign and flow interactions as "touches".
+      - name: occurred_on
+        description: Calendar date (UTC) on which the event occurred.
+      - name: touch_id
+        description: Coalescing of the Klaviyo-attributed campaign_id and flow_id.
+      - name: last_touch_at
+        description: >
+          Timestamp of when, relative to the current event, this person last interacted with a campaign or flow 
+          according to Klaviyo. This will be null if the event is not attributed to any flow or campaign. 
+      - name: last_touch_event_type
+        description: >
+          The type of event through which, relative to the current event, the person last interacted with a campaign 
+          or flow. This information is used to determine which lookback window to use (email vs sms).
+          This will be null if the event is not attributed to any flow or campaign.
+      - name: last_touch_type 
+        description: What kind of touch the event was attributed to by the package -- 'campaign', 'flow', or null. 
+      - name: last_touch_campaign_id
+        description: Foreign key referencing the CAMPAIGN that the event is attributed to by the package. 
+      - name: last_touch_flow_id
+        description: Foreign key referencing the FLOW that the event is attributed to by the package. 
+      - name: campaign_name
+        description: A name for this campaign. If not specified, this will default to the subject of the campaign.
+      - name: campaign_type
+        description: Type of campaign.
+      - name: campaign_subject_line
+        description: Type of campaign.
+      - name: flow_name
+        description: Name of the flow. 
+      - name: person_city
+        description: City that the person who triggered this event lives in. 
+      - name: person_country
+        description: Country that the person who triggered this event lives in. 
+      - name: person_region
+        description: Region or state that the person who triggered this event lives in. 
+      - name: person_email
+        description: The email address and the unique identifier for the person who triggered the event. 
+      - name: person_timezone
+        description: Timezone that the person who triggered this event is situated in. 
+      - name: integration_name
+        description: Name of the platform that triggered the event (either Klaviyo, the API, or another integration).
+      - name: integration_category
+        description: Use-case category of the platform that sent the event. 
+
+  - name: klaviyo__person_campaign_flow
```

Comment:
All `sum_` and `count_` metric fields are not documented within the dbt docs. Could we possibly add them and do a generic doc macro (similar to what Dylan typically uses) that just states this is a generated macro calculated by the package and is set within your defined variables? This way the dbt docs for this model doesn't have a large number of non documented fields.

But also understand if that is not necessary as you did a fantastic job documenting this in the README and throughout the models.

Reply:
yeah the thing is these columns are generated dynamically based on the `klaviyo__count_metrics` and `klaviyo__sum_revenue_metrics`. however, we do have default values for these variables, so maybe it would be worth documenting those? but if someone removes any of the default metrics and generates their own docs, the default metrics they removed in their own project would show up